share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```
@@ -709,3 +709,32 @@ tutorial-protected-build:
   needs:
     - artifacts: True
       job: tutorial-protected-generate
+
+########################################
+# Machine Learning
+########################################
+.ml:
+  variables:
+    SPACK_CI_STACK_NAME: ml
+
+ml-pr-generate:
+  extends: [ ".ml", ".pr-generate"]
+
+ml-protected-generate:
+  extends: [ ".ml", ".protected-generate"]
```

These jobs need to run in the same image and architecture you use in the generated pipeline, which is configured in the spack.yaml. Based on the config there:
```suggestion
.ml-generate:
  image: ghcr.io/spack/e4s-amazonlinux-2:v2022-03-21
  tags: ["spack", "aws", "public", "medium", "x86_64_v4"]

ml-pr-generate:
  extends: [ ".ml-generate", ".pr-generate"]

ml-protected-generate:
  extends: [ ".ml-generate", ".protected-generate"]
```